tests/ui/fn_params_excessive_bools.rs

```diff
@@ -24,6 +25,8 @@ struct S;
 trait Trait {
     fn f(_: bool, _: bool, _: bool, _: bool);
```

Comment:
This no longer produces a warning. My guess is that trait functions like this one don't run `check_fn` as that function requires a body. That might also be the reason, why the previous implementation iterated over the members of a module/impl block.

You can take a look at [check_trait_item](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_lint/passes/trait.LateLintPass.html#method.check_trait_item). That might help. Otherwise, you might have to iterate over the elements of a trait declaration and emit the lint using [span_lint_hir](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/diagnostics/fn.span_lint_hir.html#) which emits the lint at a given HIR node :upside_down_face:

If you implement either solution, could you also add a test(in the trait) with a default body like this:

```rs
fn i(_: bool, _: bool, _: bool, _: bool) {};
```

I would expect a warning for this one :upside_down_face:

Reply:
Another nice catch and guidance, thanks!